tools/clang/include/clang/Sema/Overload.h

```diff
@@ -97,6 +97,8 @@ namespace clang {
     ICK_HLSLVector_Splat,      ///< HLSLVector/Matrix splat
     ICK_HLSLVector_Truncation, ///< HLSLVector/Matrix truncation
     ICK_HLSL_Derived_To_Base,  ///< HLSL Derived-to-base
+    ICK_HLSLRowMajorToColMajor,///< HLSL row major to column major
+    ICK_HLSLColMajorToRowMajor,///< HLSL column major to row major
```

Comment:
Same here. Why does this need an implicit conversion kind instead of just being a FlatConversion?

Reply:
It is clearer than reuse FlatConversion.
The data is changed for major change. For FlatConversion, the data is not changed, more like a bitcast.

Reply:
My understanding is that majority only impacts the orientation on load/store, not in local memory. If that is correct orientation shouldn't change layout.

(cc @tex3d)

Reply:
It is changed with this PR. :(
The reason for doing this is hack clang type codeGen to generate same IR type is ugly.
With a llvm IR type named matrix.col, put the data in row matrix will make people confuse.

Maybe we could give a better name so it is not confusing?

Reply:
I think that changing the data storage is a problem since the underlying dxil operations for matrices assume that orientation doesn't impact the layout.